akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -167,6 +167,8 @@ private[http] class HttpRequestParser(_settings: ParserSettings,
             emitRequestStart(emptyEntity(cth))
             setCompletionHandling(HttpMessageParser.CompletionOk)
             startNewMessage(input, bodyStart)
+          } else if (!method.isEntityAccepted) {
+            failMessageStart(UnprocessableEntity, s"${method.name} requests must not have an entity")
```

Comment:
I was thinking about https://github.com/akka/akka/issues/17214 which might be out of scope here.